src/Util/EdgeConnectionUtilService.php

```diff
@@ -0,0 +1,279 @@
+use Symfony\Component\Console\Style\StyleInterface;
+
+/**
+ * Defines an interface for Edge connection classes.
+ */
+class EdgeConnectionUtilService implements EdgeConnectionUtilServiceInterface {
+
+  /**
+   * The HTTP client.
+   *
+   * @var \GuzzleHttp\Client
+   */
+  protected $httpClient;
+
+  /**
+   * Constructs a EdgeRoleCreator object.
+   *
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The HTTP client.
+   */
+  public function __construct(ClientInterface $http_client) {
+    $this->httpClient = $http_client;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function createEdgeRoleForDrupal(StyleInterface $io, callable $t, string $org, string $email, string $password, string $base_url, string $role_name) {
+
+    if (!$this->isValidEdgeCredentials($io, $t, $org, $email, $password, $base_url)) {
+      return;
+    }
+
+    $does_role_exist = $this->doesRoleExist($org, $email, $password, $base_url, $role_name);
+
+    // If role does not exist, create it.
+    if (!$does_role_exist) {
+      $io->text($t('Role !role does not exist. Creating role.', ['!role' => $role_name]));
+
+      $url = $base_url . '/o/' . $org . '/userroles';
+      try {
+        $this->httpClient->post($url, [
+          'body' => json_encode([
+            'role' => [$role_name],
+          ]),
+          'auth' => [$email, $password],
+          'headers' => [
+            'Accept' => 'application/json',
+            'Content-Type' => 'application/json',
+          ],
+        ]);
+      }
+      catch (TransferException $exception) {
+        $this->handleHttpClientExceptions($exception, $io, $t, $url, $org, $email);
+        return;
+      }
+    }
+    else {
+      $io->text('Role ' . $role_name . ' already exists.');
+    }
+
+    $this->setDefaultPermissions($io, $t, $org, $email, $password, $base_url, $role_name);
+
+    $io->success($t('Role !role is configured.  Log into apigee.com to assign a user to this role.', ['!role' => $role_name]));
+  }
+
+  /**
+   * Set default permissions for a role used for Drupal portal connections.
+   *
+   * @param \Symfony\Component\Console\Style\StyleInterface $io
+   *   The IO interface of the CLI tool calling the method.
+   * @param callable $t
+   *   The translation function akin to t().
+   * @param string $org
+   *   The Edge org to create the permissions in.
+   * @param string $email
+   *   The email of an Edge user with org admin role to make Edge API calls.
+   * @param string $password
+   *   The password of an Edge user email to make Edge API calls.
+   * @param string $base_url
+   *   The base url of the Edge API.
+   * @param string $role_name
+   *   The role name to add the permissions to.
+   */
+  protected function setDefaultPermissions(StyleInterface $io, callable $t, string $org, string $email, string $password, string $base_url, string $role_name) {
+    $io->text('Setting permissions on role ' . $role_name . '.');
+
+    $permissions = [
+      // GET access.
```

Comment:
Here you should rather describe in comments why we grant only `GET` or only `GET & PUT` or `GET, PUT & DELETE` access to certain API endpoints. These comments could provide more value than the current ones for later when someone needs to change something on this permission matrix.

Reply:
Done